Mathlib/Geometry/Manifold/ContMDiff/Basic.lean

```diff
@@ -158,6 +158,25 @@ theorem contMDiffWithinAt_id : ContMDiffWithinAt I I n (id : M → M) s x :=
 
 end id
 
+/-! ### Iterated functions -/
+
+section Iterate
+
+/-- The iterates of `C^n` functions on domains are `C^n`. -/
+theorem ContMDiffOn.iterate {f : M → M} (hf : ContMDiffOn I I n f s)
+    (hmaps : Set.MapsTo f s s) (k : ℕ) :
+    ContMDiffOn I I n (f^[k]) s := by
+  induction' k with k h
+  · simpa using contMDiffOn_id
+  · simpa using h.comp hf hmaps
+
+/-- The iterates of `C^n` functions are `C^n`. -/
+theorem ContMDiff.iterate {f : M → M} (hf : ContMDiff I I n f) (k : ℕ) :
+    ContMDiff I I n (f^[k]) :=
+  contMDiffOn_univ.mp ((contMDiffOn_univ.mpr hf).iterate (univ.mapsTo_univ f) k)
```

Comment:
Thanks. This proof term works fine. Just for your information, here's another way to write it --- which is more readable; I may also argue that the proof is so mathematically trivial that there's no need for this.
```
rw [contMDiffOn_univ] at hf ⊢
exact hf.iterate (univ.mapsTo_univ f) k